tests/integration/api_integration_test.py

```diff
@@ -37,5 +39,74 @@ def test_extract_endpoint(client):
     # make sure, that our result actually complies with the promised open api spec.
     output = Output.parse_obj(json.loads(response.text))
     assert output.url == input.url
+    assert isinstance(output.security, MetadataTags), "did not receive data for security extactor"
     # This should not be present, as the request to the accessibility container will fail
-    assert output.meta.accessibility is None, "received accessibility result but container should not be running"
+    assert isinstance(output.accessibility, Error), "received accessibility result but container should not be running"
+
+
+def test_extract_endpoint_cache(client):
+    path = Path(__file__).parent.parent / "splash-response-google.json"
+    with open(path, "r") as f:
+        splash = SplashResponse.parse_obj(json.load(f))
+
+    response = client.post(
+        "/extract_meta",
+        data=Input(url="https://google.com", splash_response=splash).json(),
+        timeout=10,
+        headers={"Cache-Control": "only-if-cached"},
+    )
+    assert response.status_code == 400, "response cannot be cached, as the whole har body is part of the request."
+
+    with mock.patch("core.website_manager.WebsiteData.fetch_content", new=AsyncMock(side_effect=lambda _: splash)):
+        response = client.post(
+            "/extract_meta",
+            data=Input(url="https://google.com").json(),
+            timeout=10,
+            headers={"Cache-Control": "only-if-cached"},
+        )
+        assert response.status_code == 404, "response can be cached but is not present in cache."
+
+        response = client.post(
+            "/extract_meta",
+            data=Input(url="https://google.com").json(),
+            timeout=10,
+            headers={"Cache-Control": "max-age=5000"},
+        )
+        assert response.status_code == 200
+
+        output = Output.parse_obj(json.loads(response.text))
+        assert output.url == "https://google.com"
+        assert isinstance(output.security, MetadataTags), "did not receive data for security extractor"
+
+        assert (
+            "https://google.com" not in client.app.cache_backend._store
+        ), "should not be cached, as it was not a complete result"
+
+        with mock.patch(
+            "features.accessibility.Accessibility._execute_api_call",
+            new=AsyncMock(side_effect=lambda website_data, session, strategy: 0.1337),
```

Comment:
What is this strategy for?

Reply:
strategy in general is either "mobile" or "desktop". For the mock here it does not matter  - but the lambda needs to have the same signature as the original function,

Reply:
Sorry, my reply here from yesterday got lost: if strategy is either mobile or desktop, then it should not be a float - could you replace the float please with a suitable string?

Reply:
you got something wrong here! The `_execute_api_call` function that gets mocked takes an strategy argument that must be either "desktop" or "mobile" but returns the score which is a float. 

The async mock here has to have this exact signature, but instead of issuing a request to the lighthouse container, it simply returns 0.1337 as a result allowing us test the accessibility extractor as if the container was actually running.